classroom.rb

```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    student&.classroom = self
+    @students << student
+  end
```

Comment:
- [ ] Imagine you're running a classroom, 🏫 and you've got a list of awesome students waiting to join. 🤩 Now, here's the smart way to do it:

1. 📝 First, you add the student's name to your student roster. This is like making sure their name is on the list of cool kids in your class, no matter what happens next. 📋
2. 🏠 Then, you officially welcome the student into your classroom by assigning them to it. 🤝 This is like saying, "Hey there, you're officially part of our classroom now!" 🎉

So, by first adding the student to the roster, you ensure they're part of your classroom gang, whether or not the official assignment process goes smoothly. It's like a fail-safe to make sure no one gets left out. 🚀

```suggestion
  def add_student(student)
    @students << student
    student.classroom = self
  end
```